app/controllers/agents_controller.rb

```diff
@@ -208,6 +208,12 @@ def complete
     render json: @agent.complete_option(params[:attribute])
   end
 
+  def destroy_undefined
+    current_user.undefined_agents.destroy_all
```

Comment:
Why does this work?  I'd think it'd look for each class as it looped over the agents to destroy them? I'm surprised you didn't need something like 
```ruby
current_user.undefined_agents.update_all(type: 'Agent');
current_user.undefined_agents.destroy_all
```